app/services/category_transactions/creator.rb

```diff
@@ -38,7 +38,7 @@ def category
   end
 
   def valid?
-    user.free_balance >= @amount
+    true if user.free_balance >= @amount && @amount > 0
```

Comment:
@marderer it will return `true` or `false` because of comparison, so, there is no need in returning `true` directly